eclair-core/src/test/scala/fr/acinq/eclair/router/GraphSpec.scala

```diff
@@ -255,10 +256,70 @@ class GraphSpec extends AnyFunSuite {
+    val path = yenKshortestPaths(graph, a, e, 100000000 msat,
+      Set.empty, Set.empty, Set.empty, 2,
       Right(HeuristicsConstants(1.0E-8, RelayFees(2000 msat, 500), RelayFees(50 msat, 20), useLogProbability = true)),
       BlockHeight(714930), _ => true, includeLocalChannelCost = true)
-    assert(path.path == Seq(edgeAB, edgeBC, edgeCE))
+    assert(path.length == 2)
+    assert(path.head.path == Seq(edgeAB, edgeBC, edgeCE))
+  }
+
+  test("amount with fees less than channel capacity for C->D, but not C->E") {
+    /*
+    The channel C -> D is large enough for the payment and associated fees to go through, but C -> E is not.
+    A --> B --> C <-> D
+                 \   /
+                  \ /
+                   E
+    */
+    val edgeAB = makeEdge(1L, a, b, 10001 msat, 0, capacity = 200000 sat)
+    val edgeBC = makeEdge(2L, b, c, 10000 msat, 0, capacity = 200000 sat)
+    val edgeCD = makeEdge(3L, c, d, 20001 msat, 0, capacity = 100011 sat)
+    val edgeDC = makeEdge(4L, d, c, 1 msat, 0, capacity = 300000 sat)
+    val edgeCE = makeEdge(5L, c, e, 10 msat, 0, capacity = 10000 sat)
+    val edgeDE = makeEdge(6L, d, e, 9 msat, 0, capacity = 200000 sat)
+    val graph = DirectedGraph(Seq(edgeAB, edgeBC, edgeCD, edgeDC, edgeCE, edgeDE))
+
+    val paths = yenKshortestPaths(graph, a, e, 90000000 msat,
+      Set.empty, Set.empty, Set.empty, 2,
+      Right(HeuristicsConstants(1.0E-8, RelayFees(2000 msat, 500), RelayFees(50 msat, 20), useLogProbability = true)),
+      BlockHeight(714930), _ => true, includeLocalChannelCost = true)
+    // Even though paths to find is 2, we only find 1 because that is all the valid paths that there are.
+    assert(paths.length == 1)
+    assert(paths.head.path == Seq(edgeAB, edgeBC, edgeCD, edgeDE))
+  }
+
+  /**
+   * Find all the shortest paths using the example described in
+   * https://en.wikipedia.org/wiki/Yen%27s_algorithm#:~:text=Yen's%20algorithm
+   */
+  test("all shortest paths are found") {
+    //  There will be 3 shortest paths
+    //  C --> D --> F
+    //   \    ^    /|\
+    //    \   |  /  | \
+    //     \ | /    |  \
+    //       E----->G-->H
+    val edgeCD = makeEdge(1L, c, d, 10 msat, 0, capacity = 301000 sat, minHtlc = 1000 msat)
+    val edgeDF = makeEdge(2L, d, f, 10 msat, 0, capacity = 401000 sat, minHtlc = 1000 msat)
+    val edgeCE = makeEdge(3L, c, e, 10 msat, 0, capacity = 201000 sat, minHtlc = 1000 msat)
+    val edgeED = makeEdge(4L, e, d, 10 msat, 0, capacity = 101000 sat, minHtlc = 1000 msat)
+    val edgeEF = makeEdge(5L, e, f, 10 msat, 0, capacity = 201000 sat, minHtlc = 1000 msat)
+    val edgeFG = makeEdge(6L, f, g, 10 msat, 0, capacity = 201000 sat, minHtlc = 1000 msat)
+    val edgeFH = makeEdge(7L, f, h, 10 msat, 0, capacity = 101000 sat, minHtlc = 1000 msat)
+    val edgeEG = makeEdge(8L, e, g, 10 msat, 0, capacity = 301000 sat, minHtlc = 1000 msat)
+    val edgeGH = makeEdge(9L, g, h, 10 msat, 0, capacity = 201000 sat, minHtlc = 1000 msat)
```

Comment:
This is absolutely not the wikipedia example. You matched the costs from the wikipedia example to the capacities but the capacities are actually inversely correlated with the cost.